projects/packages/videopress/src/client/admin/components/edit-video-details/use-edit-details.ts

```diff
@@ -9,51 +9,139 @@ import { useNavigate } from 'react-router-dom';
  * Internal dependencies
  */
 import useMetaUpdate from '../../../hooks/use-meta-update';
+import usePosterImage from '../../../hooks/use-poster-image';
+import usePosterUpload from '../../../hooks/use-poster-upload';
 import { STORE_ID } from '../../../state';
 import useVideo from '../../hooks/use-video';
 
+const usePosterEdit = ( { video } ) => {
+	const [ videoFrameMs, setVideoFrameMs ] = useState( null );
+	const [ currentTime, setCurrentTime ] = useState( null );
+	const [ frameSelectorIsOpen, setFrameSelectorIsOpen ] = useState( false );
+
+	const posterUpload = usePosterUpload( video?.guid );
+	const posterImage = usePosterImage( video?.guid );
+
+	const posterImagePooling = ( onGenerate = null ) => {
+		posterImage().then( ( { data: result } ) => {
+			if ( result?.generating ) {
+				setTimeout( () => posterImagePooling( onGenerate ), 2000 );
+			} else if ( result?.poster ) {
+				onGenerate?.( result?.poster );
+			}
+		} );
```

Comment:
Probably needs a `.catch`.

Reply:
Yep, I left a comment here: https://github.com/Automattic/jetpack/blob/2aa081d70e4b5a9034664caf639bf1380ad39660/projects/packages/videopress/src/client/admin/components/edit-video-details/use-edit-details.ts#L134 ... that we need to handle errors